apps/www/src/components/Chat.tsx

```diff
@@ -0,0 +1,85 @@
+import { Message } from "@/components/demo/message/Message";
+import { Button } from "@/components/ui/Button";
+import { Input } from "@/components/ui/Input";
+import { Textarea } from "@/components/ui/Textarea";
+
+import * as React from "react";
+
+export const Chat = () => {
+  const [userContent, setUserContent] = React.useState<string>("");
+  const [currentId, setCurrentId] = React.useState<number>();
+  const [currentApiKey, setCurrentApiKey] = React.useState<string>();
+  const [output, setOutput] = React.useState<string>("");
+  const [isStreamFinished, setIsStreamFinished] =
+    React.useState<boolean>(false);
+
+  React.useEffect(() => {
+    setOutput("");
+    if (currentId && currentApiKey) {
+      const eventSource = new EventSource(
+        `/api/chat/openai?id=${currentId}&apiKey=${currentApiKey}`,
+      );
+
+      eventSource.addEventListener("error", () => eventSource.close());
+
+      eventSource.addEventListener("token", (e) => {
+        // avoid newlines getting messed up
+        const token = e.data.replaceAll("$NEWLINE$", "\n");
+        setOutput((prevResponse) => `${prevResponse}${token}`);
+      });
+
+      eventSource.addEventListener("finished", (e) => {
+        console.log("finished", e);
+        eventSource.close();
+        setIsStreamFinished(true);
+      });
+
+      () => eventSource.close();
+    }
+  }, [currentId, currentApiKey]);
+
+  const handleUpdateUserContent = (
+    e: React.ChangeEvent<HTMLTextAreaElement>,
+  ) => {
+    const newUserContent = e.target.value;
+    setUserContent(newUserContent);
+  };
+
+  const handleUpdateApiKey = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newApiKey = e.target.value;
+    setCurrentApiKey(newApiKey);
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
```

Comment:
Probably needs validation for API key and maybe prompt (not as big a deal as it will just give a boring response).

Reply:
validation for API key would be good 👍